src/psyclone/psyGen.py

```diff
@@ -2433,6 +2431,39 @@ def gen_code(self, parent):
             for call in reprod_red_call_list:
                 call.reduction_sum_loop(parent)
 
+    def begin_string(self):
+        '''Returns the beginning statement of this directive, i.e.
+        "omp parallel". The visitor is responsible for adding the
+        correct directive begin.
+
+        :returns: the beginning statement oforthis directive.
+        :rtype: str
+        '''
+
+        # pylint: disable=no-self-use
+        result = "omp parallel"
+        # TODO: not yet working with NEMO
+        # if not self._reprod:
+        #     result += self._reduction_string()
+        private_list = self._get_private_list()
+        private_str = self.list_to_string(private_list)
+
+        if private_str:
+            result = "{0} private({1})".format(result, private_str)
+        return result
+
+    def end_string(self):
+        '''Returns the end (or closing) statement of this directive, i.e.
+        "omp end parallel". The visitor is responsible for adding the
+        correct directive begin.
+
+        :returns: the end statement for this directive.
+        :rtype: str
+        '''
```

Comment:
Blank line inside docstring rather than after it please.

Reply:
Done.